internal/api/metrics/v1/http.go

```diff
@@ -155,6 +158,36 @@ func NewHandler(read, write *url.URL, opts ...HandlerOption) http.Handler {
 				proxyRemoteWrite,
 			))
 		})
+		return r
+	}
+
+	if write != nil {
```

Comment:
Is there any case where `write` is nil? Is it worth including this as a check?

Reply:
yes it could be an option. unlikely and for acm specific use case it wont be nil, but we just check it before registering the handler